endpoints/openrtb2/amp_auction.go

```diff
@@ -298,12 +344,14 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 	ao.AmpTargetingValues = targets
 
 	// add debug information if requested
-	if reqWrapper.Test == 1 && eRErr == nil {
-		if extResponse.Debug != nil {
-			ampResponse.Debug = extResponse.Debug
-		} else {
-			glog.Errorf("Test set on request but debug not present in response: %v", err)
-			ao.Errors = append(ao.Errors, fmt.Errorf("Test set on request but debug not present in response: %v", err))
+	if reqWrapper != nil {
+		if reqWrapper.Test == 1 && eRErr == nil {
```

Comment:
You can remove an `if` statement by combining the top two conditions:
`if reqWrapper != nil  && reqWrapper.Test == 1 && eRErr == nil {`